webui/react/src/components/UPlotChart.tsx

```diff
@@ -19,29 +19,30 @@ const SCROLL_THROTTLE_TIME = 500;
 const UPlotChart: React.FC<Props> = ({ data, options }: Props) => {
   const [ chart, setChart ] = useState<uPlot>();
   const chartDivRef = useRef<HTMLDivElement>(null);
+  const hasData: boolean = useMemo(() => !!(data && data[0].length > 0), [ data ]);
 
   /*
    * Chart setup.
    */
   useEffect(() => {
-    if (!chartDivRef.current || !options) return;
+    if (!chartDivRef.current || !hasData || !options) return;
 
     const optionsExtended = uPlot.assign(
       {
         cursor: { drag: { dist: 5, uni: null, x: true, y: true } },
+        hooks: { ready: [ (chart: uPlot) => setChart(chart) ] },
         width: chartDivRef.current.offsetWidth,
       },
       options,
     );
 
     const plotChart = new uPlot(optionsExtended as uPlot.Options, [ [] ], chartDivRef.current);
-    setChart(plotChart);
```

Comment:
We might `setChart` and trigger the other `useEffect` before the chart is ready.
Moved to the ready hook.